### PR TITLE
add base run functionality

### PR DESCRIPTION
**Pull request by ulya-tkch**
This PR adds Python API code for the base run functionality as well as wait until cleanset ready options.

### CODE REVIEW

File: cleanlab_studio/studio/enrichment.py [hunk 1]
```diff
@@ -155,6 +150,58 @@ def preview(
...
+        *,
+        new_column_name: str,
+    ) -> EnrichmentResult:
+        """Enrich the full dataset."""
+        if self._run_called:
+            raise ValueError(RUN_ALREADY_CALLED_ERROR_MESSAGE)
+        else:
+            self._run_called = True
+
+        # TODO: #1.3 Error is thrown if {new_column_name} or {new_column_name}_log or {new_column_name}_trustworthiness_score exist in dataset
```
Comment (ulya-tkch) [inline cleanlab_studio/studio/enrichment.py:165]: No need to do this logic here, done in backend

File: cleanlab_studio/studio/enrichment.py [hunk 2]
```diff
@@ -155,6 +150,58 @@ def preview(
...
+            extraction_pattern=extraction_pattern,
+            optimize_prompt=options.get("optimize_prompt", True),
+            prompt=options["prompt"],
+            quality_preset=options.get("quality_preset", "medium"),
+            replacements=replacements,
+            tlm_options=cast(Dict[str, Any], options.get("tlm_options"))
+            if options.get("tlm_options")
+            else {},
+        )
+        epr = EnrichmentResult.from_dict(response)
```
Comment (ulya-tkch) [inline cleanlab_studio/studio/enrichment.py:192]: full name var

File: cleanlab_studio/errors.py [hunk 3]
```diff
@@ -131,6 +135,10 @@ def __str__(self) -> str:
         return error_msg
 
 
+class EnrichmentProjectHandledError(CleansetHandledError):
```
Comment (ulya-tkch) [inline cleanlab_studio/errors.py:138]: Don't subclass from CleansetHandledError

File: cleanlab_studio/internal/enrich_helpers.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+from cleanlab_studio.internal.api import api
+from cleanlab_studio.internal.util import log_internal_error
+
+
+from tqdm import tqdm
+
+
+import itertools
+import time
+import traceback
```
Comment (eytanhanig) [inline cleanlab_studio/internal/enrich_helpers.py:11]: nit: It looks like isort didn't run on this file. Do you have pre-commit installed for this repo?

File: cleanlab_studio/internal/api/api.py [hunk 5]
```diff
@@ -673,6 +684,41 @@ def enrichment_preview(
...
 
+def run(
+    api_key: str,
+    new_column_name: str,
+    project_id: str,
+    prompt: str,
+    constrain_outputs: Optional[List[str]] = None,
+    extraction_pattern: Optional[str] = None,
+    optimize_prompt: Optional[bool] = True,
+    quality_preset: Optional[TLMQualityPreset] = "medium",
```
Comment (eytanhanig) [inline cleanlab_studio/internal/api/api.py:695]:
```suggestion
    quality_preset: TLMQualityPreset = "medium",
```

**Review by eytanhanig — APPROVED** [review]

File: cleanlab_studio/internal/api/api.py (hunk 5 above)
Comment (ulya-tkch) [reply cleanlab_studio/internal/api/api.py:695]: Why are we making quality_preset not optional here if it is optional in `enrichment_preview()` method?

**Pull request closed without merging**